daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/unparsers/runtime1/BinaryNumberUnparsers.scala

```diff
@@ -268,6 +306,32 @@ abstract class BinaryDecimalUnparserBase(
     nBits: Int,
     finfo: FormatInfo
   ): Boolean = {
+    val state = finfo.asInstanceOf[UState]
+    val isSigned = signed == Yes
+    val minWidth = if (isSigned) 2 else 1
+    if (nBits < minWidth) {
+      val signedStr = if (isSigned) "a signed" else "an unsigned"
+      val outOfRangeFmtStr =
+        "Minimum length for %s binary decimal is %d bit(s), number of bits %d out of range. " +
+          "An unsigned decimal with length 1 bit could be used instead."
+      if (isSigned && state.tunable.allowSignedIntegerLength1Bit && nBits == 1) {
+        state.SDW(
+          WarnID.SignedBinaryIntegerLength1Bit,
+          outOfRangeFmtStr,
+          signedStr,
+          minWidth,
+          nBits
+        )
+      } else {
+        UE(
+          state,
+          outOfRangeFmtStr,
+          signedStr,
+          minWidth,
+          nBits
+        )
```

Comment:
I think this same chunk of code with very minor variations appears in multiple places. Can we add trait to avoid all this duplication?